crates/sui-indexer-restorer/src/snapshot.rs

```diff
@@ -0,0 +1,214 @@
+                    "[{elapsed_precise}] {wide_bar} {pos} out of {len} move object files restored ({msg})",
+                )
+                .unwrap(),
+            ),
+        ));
+
+        Abortable::new(
+            async move {
+                let sema_limit = Arc::new(Semaphore::new(concurrency));
+                let mut restore_tasks = vec![];
+
+                for (bucket, (part_num, file_metadata)) in input_files.into_iter() {
+                    let sema_limit_clone = sema_limit.clone();
+                    let epoch_dir_clone = epoch_dir.clone();
+                    let remote_object_store_clone = remote_object_store.clone();
+                    let sha3_digests_clone = sha3_digests.clone();
+                    let bar_clone = move_object_progress_bar.clone();
+                    let db = self.db.clone();
+                    let next_cp = self.next_checkpoint_after_epoch;
+
+                    let restore_task = task::spawn(async move {
+                        let mut conn = db.connect().await?;
+                        let _permit = sema_limit_clone.acquire().await.unwrap();
+                        let object_file_path = file_metadata.file_path(&epoch_dir_clone);
+                        let (bytes, _) = download_bytes(
+                            remote_object_store_clone,
+                            &file_metadata,
+                            epoch_dir_clone,
+                            sha3_digests_clone,
+                            &&bucket,
+                            &part_num,
+                            Some(512),
+                        )
+                        .await;
+                        info!(
+                            "Finished downloading move object file {:?}",
+                            object_file_path
+                        );
+                        let mut object_infos = vec![];
+                        let _result: Result<(), anyhow::Error> =
+                            LiveObjectIter::new(&file_metadata, bytes.clone()).map(|obj_iter| {
+                                for object in obj_iter {
+                                    match object {
+                                        LiveObject::Normal(obj) => {
+                                            let object_info = create_stored_obj_info(
+                                                obj.id(),
+                                                next_cp as i64,
+                                                &obj,
+                                            );
+                                            object_infos.push(object_info);
+                                        }
+                                        LiveObject::Wrapped(_) => {}
+                                    }
+                                }
+                            });
+                        let object_infos =
+                            object_infos.into_iter().collect::<Result<Vec<_>, _>>()?;
```

Comment:
Quite confused by this code -- how can it produce an error, and if it can, why are we ignoring it? It seems like the right thing to do would be to use `flat_map` on the outside, and then `filter_map` on the inside? Something like:

```suggestion
                        let object_infos = LiveObjectIter::new(&file_metadata, bytes.clone())
                            .flat_map(|obj_iter| {
                                obj_iter.filter_map(|obj| match object {
                                    LiveObject::Wrapped(_) => None,
                                    LiveObject::Normal(obj) => Some(StoredObjectInfo::try_from(
                                        obj.id(),
                                        next_cp as i64,
                                        &obj,
                                    ))
                                })
                            })
                            .collect::<Result<Vec<_>>>()?;
```

Reply:
the error is from LiveObjectIter::new() 
https://github.com/gegaowp/sui/blob/indexer-restore-init/crates/sui-snapshot/src/reader.rs#L638
and it should not be ignored indeed, changed now.